packages/workbox-build/src/lib/runtime-caching-converter.js

```diff
@@ -60,7 +60,40 @@ function getOptionsString(options = {}) {
       throw new Error(`${errors['bad-runtime-caching-config']} ${pluginName}`);
     }
 
-    plugins.push(`new ${pluginString}(${JSON.stringify(pluginConfig)})`);
+    let pluginCode;
+    switch (pluginName) {
+      // Special case logic for plugins that have a required parameter, and then
+      // an additional optional config parameter.
+      case 'backgroundSync': {
```

Comment:
I was thinking of combining this and the next `case`, which would prevent some code duplication. One reads the required value from the `name` parameter, and the other reads it from `channelName`, though, so there would be some extra logic that would need to be added.

I ended up keeping them separate, but if folks would prefer them combined, I could do that.

Reply:
I think seperate makes sense - makes the tests super easy to read.